mycroft/client/speech/mic.py

```diff
@@ -303,11 +305,22 @@ def _wait_until_wake_word(self, source, sec_per_buffer):
         counter = 0
 
         while not said_wake_word:
-            if check_for_signal('buttonPress'):
-                logger.debug("Button Pressed, wakeword not needed")
+            # Check if told programatically to skip the wake word for dialog
+            if check_for_signal('startListening'):
                 said_wake_word = True
                 continue
 
+            # Pressing the button can start recording (unless it is being
+            # used to mean 'stop' instead)
+            if check_for_signal('buttonPress', 1):
+                # give others time to consume this as 'stop'
+                sleep(0.25)
+                if check_for_signal('buttonPress'):
+                    # signal is still here, assume it was to begin recording
+                    logger.debug("Button Pressed, wakeword not needed")
+                    said_wake_word = True
+                    continue
+
```

Comment:
I added the "startListening" to make it clear that the request was from the framework and explicitly about listening, not to be confused with the consumption of the buttonPress.